mono/metadata/threads.c

```diff
@@ -4993,9 +4993,12 @@ mono_threads_join_threads (void)
 		}
 		joinable_threads_unlock ();
 		if (found) {
-			if (thread != pthread_self ())
+			if (thread != pthread_self ()) {
```

Comment:
done in https://github.com/mono/mono/pull/3074/commits/a9675a64b0d3cd319dc6811be55a220ace3cae49